### ISSUE TITLE
RemoteFunction not testable

### ISSUE THREAD

**Issue by MaxE17677** [gkralik/php7-sapnwrfc#133]
### Bug Report

| Q                         | A     |
| ------------------------- | ----- |
| PHP version               |7.4.29, 7.4.14, 8.1.7 |
| SAP NW RFC SDK version    | nwrfc750P_9-70002752 |
| Extension version/git ref | 2.0.0-beta2 |

#### Summary

1. remotefunction cannot be mocked because of the "final" flag
2. unable to destroy self-created RemoteFunction objects because of "Segmentation Fault" `interrupted by signal 11: SIGSEGV`.

#### Actual behavior

At the beginning: Thank you very much for the terrific work. With a real connection and a RemoteFunction object created by this connection everything works as expected.

But to simulate the behavior without sending and receiving data with a real connection, it would be nice to override or mock up the RemoteFunction.

The final class RemoteFunction cannot be emulated because of **final**. 
I removed the line `sapnwrfc_function_ce->ce_flags |= ZEND_ACC_FINAL` and the following code works
```php
$function = $this->createMock(\SAPNWRFC\RemoteFunction::class);
$function->method('invoke')->with($data)->willReturn($result);
```
But the test doesn't work anymore if the variables are " freed". It leads to a "segmentation fault".

Examples to reproduce:

```php
$f = new \SAPNWRFC\RemoteFunction();
unset($f);
```
or in a test
```php
$function = $this->createMock(\SAPNWRFC\RemoteFunction::class);
unset($function);
```

#### 

```php
use App\SAP\Dto\StockData;
use App\SAP\Service\Client;
use Codeception\Test\Unit;
use SAPNWRFC\Connection;
use SAPNWRFC\RemoteFunction;
use Symfony\Component\Serializer\Normalizer\NormalizerInterface;

Class CientTest extends Unit
{
    public function testTransferStock (): void
    {
        $data = new stockData();
        $normalizedData = [...];
        $result = ['some' => 'result'];
         
        $normalizer = $this->createMock(NormalizerInterface::class);
        $connection = $this->createMock(Connection::class);
        $function = $this->createMock(RemoteFunction::class);
         
        $normalizer->method('normalize')->willReturn($normalizedData);
        $connection->method('getFunction')->willReturn($function);
        $function->method('invoke')->with($data)->willReturn($result);
         
        $classUnderTest = new Client($connection, $normalizer);

        $transferStock = $classUnderTest->transferStock($data);
        self::assertSame($result, $transferStock); 
    }
}
```

#### Expected behavior

The free method for destructuring self-created RemoteFunction objects does not result in a segmentation fault.

**Comment by gkralik** [comment]
Thanks, I'll have a look in the coming days.

**Comment by gkralik** [comment]
This should be fixed with 7a2a033 "Allow mocking RemoteFunction"
